aws/resource_aws_lambda_layer_version_test.go

```diff
@@ -0,0 +1,369 @@
+	layerName := fmt.Sprintf("tf_acc_lambda_layer_runtimes_%s", rString)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLambdaLayerVersionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLambdaLayerVersionCompatibleRuntimes(layerName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsLambdaLayerVersionExists(resourceName, layerName),
+					resource.TestCheckResourceAttr(resourceName, "compatible_runtimes.#", "2"),
+				),
+			},
+
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"filename"},
+			},
+		},
+	})
+}
+
+func TestAccAWSLambdaLayerVersion_description(t *testing.T) {
+	resourceName := "aws_lambda_layer_version.lambda_layer_test"
+	rString := acctest.RandString(8)
+	layerName := fmt.Sprintf("tf_acc_lambda_layer_description_%s", rString)
+	testDescription := "test description"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLambdaLayerVersionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLambdaLayerVersionDescription(layerName, testDescription),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsLambdaLayerVersionExists(resourceName, layerName),
+					resource.TestCheckResourceAttr(resourceName, "description", testDescription),
+				),
+			},
+
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"filename"},
+			},
+		},
+	})
+}
+
+func TestAccAWSLambdaLayerVersion_licenseInfo(t *testing.T) {
+	resourceName := "aws_lambda_layer_version.lambda_layer_test"
+	rString := acctest.RandString(8)
+	layerName := fmt.Sprintf("tf_acc_lambda_layer_license_info_%s", rString)
+	testLicenseInfo := "MIT"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLambdaLayerVersionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSLambdaLayerVersionLicenseInfo(layerName, testLicenseInfo),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsLambdaLayerVersionExists(resourceName, layerName),
+					resource.TestCheckResourceAttr(resourceName, "license_info", testLicenseInfo),
+				),
+			},
+
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"filename"},
+			},
+		},
+	})
+}
+
+func testAccCheckLambdaLayerVersionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).lambdaconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_lambda_layer" {
```

Comment:
This should be `aws_lambda_layer_version`

```suggestion
		if rs.Type != "aws_lambda_layer_version" {
```

Reply:
whoops, missed that 🤦‍♂️